File: fib/fib.rb
def fib(n)
  return n if n <= 1
  fib(n - 1) + fib(n - 2)
end

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
result = fib(40)
elapsed = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start

puts "Result: #{result}"
puts "Time: #{'%.3f' % elapsed}s"

File: fib/fib.py
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

start = time.monotonic()
result = fib(40)
elapsed = time.monotonic() - start

print(f"Result: {result}")
print(f"Time: {elapsed:.3f}s")
